### PR TITLE
Add 'updateMavenProject' method into MavenProject

### LINKED ISSUE #1876: Allow to call "Update Maven Project" on Maven project
MavenProject class should offer 'updateMavenProject' method. "Update Maven Project" should be represented by subclass of TitleAreaDialog.

### PR DESCRIPTION
**Pull request by luvalach**
Enhance MavenProject class by addding 'updateMavenProject' method. Add UpdateMavenProjectsDialog and MavenProjectTest. (fixes #1876)

Signed-off-by: Lukáš Valach <lvalach@redhat.com>

### CODE REVIEW

**Comment by genie-reddeer** [comment]
Can one of the admins verify this patch?

**Comment by rawagner** [comment]
testPR

File: plugins/org.eclipse.reddeer.eclipse/src/org/eclipse/reddeer/eclipse/m2e/core/ui/internal/dialogs/UpdateMavenProjectsDialog.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+	public static final String CLEAN_PROJECTS = "Clean projects";
+
+	public UpdateMavenProjectsDialog() {
+		super(SHELL_TITLE);
+	}
+
+	/**
+	 * Click "OK" button.
+	 */
+	public void ok() {
```
Comment (rawagner) [inline plugins/org.eclipse.reddeer.eclipse/src/org/eclipse/reddeer/eclipse/m2e/core/ui/internal/dialogs/UpdateMavenProjectsDialog.java]: can you please also add method ok(TimePeriod waitForJobs). TimePeriod.LONG may not be enough for all cases

File: plugins/org.eclipse.reddeer.eclipse/src/org/eclipse/reddeer/eclipse/core/resources/MavenProject.java [hunk 2]
```diff
@@ -28,4 +30,12 @@ public MavenProject(TreeItem item) {
 	public String[] getNatureIds() {
 		return new String[] { "org.eclipse.m2e.core.maven2Nature" };
 	}
+	
+	public void updateMavenProject() {
```
Comment (rawagner) [inline plugins/org.eclipse.reddeer.eclipse/src/org/eclipse/reddeer/eclipse/core/resources/MavenProject.java]: can you please add updateMavenProject(TimePeriod waitForJobs) which would call updateProject.ok(TimePeriod). See my comment below

**Comment by luvalach** [comment]
Done, could you check it again please?

**Comment by rawagner** [comment]
testPR

**Comment by rawagner** [comment]
@luvalach  looks good, but the test is failing. Can you take a look please ? https://hudson.eclipse.org/reddeer/job/reddeer.prcheck/211/testReport/junit/org.eclipse.reddeer.eclipse.test.core.resources/MavenProjectTest%20no-configuration/org_eclipse_reddeer_eclipse_test_core_resources_MavenProjectTest_no_configuration/

**Comment by luvalach** [comment]
testPR

**Comment by rawagner** [comment]
testPR

**Comment by luvalach** [comment]
Few tests failed but it doesn't seems to be related to my changes. Could you have a look please?

**Pull request merged**